docs/tutorial/workarounds.md

```diff
@@ -0,0 +1,142 @@
+# Common tricks used in (T)FHE
+
+As explained in the [Basics of FHE](../getting-started/fhe_basics.md), the challenge for developers
+is to adapt their code to fit FHE constraints. In this document we have collected some common examples
+to illustrate the kind of optimization one can do to get better performance.
+
+{% hint style="info" %}
+All code snippets provided here are for educational purpose. In future version of Concrete, some
+functions described here could be directly available in a more generic and efficient form.
+These code snippets are coming from support answers in our [community forum](https://community.zama.ai)
+{% endhint %}
+
+
+## Minimum for Two values
+
+In this first example, we compute a minimum by creating a difference between the two numbers $$y$$ and $$x$$
+and conditionally remove this diff from $$y$$ to either get $$x$$ if $$y>x$$ or $$y$$ if $$x>y$$:
+
+```python
+def min_two(x, y):
+	diff = y - x
+	min_x_y = y - np.maximum(y - x, 0)
+	return min_x_y
+```
+
+## Maximum for Two values
+
+The companion example of above with the maximum value of two integers instead of the minimum:
+
+```python
+def max_two(x, y):
+	diff = y - x
+	max_x_y = y - np.minimum(y - x, 0)
+	return max_x_y
+```
+
+## Minimum for several values
+
+And an extension for more than two values:
+
+```python
+    def fhe_min(*args):
+        remaining = list(args)
+        while len(remaining) > 1:
+            a = remaining.pop()
+            b = remaining.pop()
+            min_a_b = b - np.maximum(b - a, 0)
+            remaining.insert(0, min_a_b)
+        return remaining[0]
+```
+
+## Retrieving a value with an encrypted index
+
+This example show how to deal with an array and an encrypted index. It will create a "selection" array filled with `0` except for the requested index that will be `1`, and sum the products of all array values by this selection array:
```

Comment:
This example showS

Reply:
the sentence is complicated: maybe say that the goal is to compute array[index], where index is encrypted and array is a dynamic encrypted array, ie not a constant table known table (which is what we have in a regular PBS)

Reply:
The sentence explains the solution.

Reply:
With @andrei-stoian-zama , we thought that this sentence was complicated to understand

Reply:
It explains the algorithm, not the purpose.

Reply:
let sort that in the language review